### PR TITLE
Cluster support

### PR DESCRIPTION
**Pull request by gustavonalle**
Add support for clustering in the image plus other small changes:

* Update to Infinispan 9.0.0.Alpha3.
   Can we tag master in order to build 9.x images and create another branch for 8.2.x ?

* Extract using unzip instead of bsdtar to preserve file permissions
* Add entrypoint script to support running clustered mode besides standalone
* Support to pass parameters to the server in order to choose the JGroups stack
* Support for domain mode
* Added usage instructions

### CODE REVIEW

**Comment by infinispanrelease** [comment]
Started - TeamCity Infinispan :: Docker image Pull requests JDK8 [Build 15](http://ci.infinispan.org/viewLog.html?buildId=41067&buildTypeId=Infinispan_DockerImagePullRequestsJdk8&guest=1) for 7c5ce8bbf8283dab1620b9b4a379798b3202ff9c

**Comment by infinispanrelease** [comment]
**FAILURE** - TeamCity Infinispan :: Docker image Pull requests JDK8 [Build 15/](http://ci.infinispan.org/viewLog.html?buildId=41067&buildTypeId=Infinispan_DockerImagePullRequestsJdk8&guest=1) for 7c5ce8bbf8283dab1620b9b4a379798b3202ff9c
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:01:36

Process exited with code 1

Error message is logged

**Comment by infinispanrelease** [comment]
Started - TeamCity Infinispan :: Docker image Pull requests JDK8 [Build 16](http://ci.infinispan.org/viewLog.html?buildId=41068&buildTypeId=Infinispan_DockerImagePullRequestsJdk8&guest=1) for 12069f3479f853eba09d15a69c7371bd5536a50f

**Comment by infinispanrelease** [comment]
**FAILURE** - TeamCity Infinispan :: Docker image Pull requests JDK8 [Build 16/](http://ci.infinispan.org/viewLog.html?buildId=41068&buildTypeId=Infinispan_DockerImagePullRequestsJdk8&guest=1) for 12069f3479f853eba09d15a69c7371bd5536a50f
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:01:03

Process exited with code 1

Error message is logged

**Comment by gustavonalle** [comment]
Updated with support for domain mode

**Comment by infinispanrelease** [comment]
**FAILURE** - TeamCity Infinispan :: Docker image Pull requests JDK8 [Build 17/gustavonalle:cluster](http://ci.infinispan.org/viewLog.html?buildId=41074&buildTypeId=Infinispan_DockerImagePullRequestsJdk8&guest=1) for a12f15b87677bb9723fd1fbc2eab50f98949a84f
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:01:16

Process exited with code 1

Error message is logged

**Comment by gustavonalle** [comment]
Updated with SHA1 calculation, fix minor script error and cosmetic changes to documentation

**Comment by infinispanrelease** [comment]
Started - TeamCity Infinispan :: Docker image Pull requests JDK8 [Build 18](http://ci.infinispan.org/viewLog.html?buildId=41119&buildTypeId=Infinispan_DockerImagePullRequestsJdk8&guest=1) for 308842d50e9653bb2af029b0c75621a36180b486

**Comment by infinispanrelease** [comment]
**UNKNOWN** - TeamCity Infinispan :: Docker image Pull requests JDK8 [Build 18](http://ci.infinispan.org/viewLog.html?buildId=41119&buildTypeId=Infinispan_DockerImagePullRequestsJdk8&guest=1) for 308842d50e9653bb2af029b0c75621a36180b486
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 03:22:54

**Comment by infinispanrelease** [comment]
Started - TeamCity Infinispan :: Docker image Pull requests JDK8 [Build 22](http://ci.infinispan.org/viewLog.html?buildId=41126&buildTypeId=Infinispan_DockerImagePullRequestsJdk8&guest=1) for 308842d50e9653bb2af029b0c75621a36180b486

**Comment by infinispanrelease** [comment]
**SUCCESS** - TeamCity Infinispan :: Docker image Pull requests JDK8 [Build 22/gustavonalle:cluster](http://ci.infinispan.org/viewLog.html?buildId=41126&buildTypeId=Infinispan_DockerImagePullRequestsJdk8&guest=1) for 308842d50e9653bb2af029b0c75621a36180b486
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:01:03

File: server/Dockerfile [hunk 1]
```diff
@@ -4,14 +4,26 @@ FROM jboss/base-jdk:8
 # Set the INFINISPAN_SERVER_HOME env variable
 ENV INFINISPAN_SERVER_HOME /opt/jboss/infinispan-server
 
+ENV INFINISPAN_SHA a55ca9cddd5b3373029b780f6530ccdffdc8396f
```
Comment (slaskawi) [inline server/Dockerfile]: Maybe we could extract it from Nexus? https://repository.jboss.org/nexus/content/repositories/public/org/infinispan/infinispan-embedded/8.2.3.Final/infinispan-embedded-8.2.3.Final.jar.sha1

File: server/Dockerfile [hunk 2]
```diff
@@ -4,14 +4,26 @@ FROM jboss/base-jdk:8
...
-# Download and unzip Infinispan server
-RUN cd $HOME && curl http://downloads.jboss.org/infinispan/$INFINISPAN_VERSION/infinispan-server-$INFINISPAN_VERSION-bin.zip | bsdtar -xf - && mv $HOME/infinispan-server-$INFINISPAN_VERSION $HOME/infinispan-server && chmod +x /opt/jboss/infinispan-server/bin/standalone.sh
+ENV MGMT_USER admin
 
-# Expose Infinispan server  ports 
-EXPOSE 8080 8181 9990 11211 11222 
+ENV MGMT_PASS admin
+
+# Server download location
+ENV DISTRIBUTION_URL https://repo1.maven.org/maven2/org/infinispan/server/infinispan-server-build/$INFINISPAN_VERSION/infinispan-server-build-$INFINISPAN_VERSION.zip
```
Comment (slaskawi) [inline server/Dockerfile]: How about using [Nexus API](https://repository.sonatype.org/nexus-restlet1x-plugin/default/docs/path__artifact_maven_redirect.html) for doing this? Here is an example:
```
wget -L "https://repository.jboss.org/nexus/service/local/artifact/maven/redirect?r=snapshots&g=org.infinispan.server&a=infinispan-server-build&v=9.0.0-SNAPSHOT&e=zip"
```

Comment (gustavonalle) [inline server/Dockerfile]: I used maven central because for me it's 100x faster to download than Nexus. Is that the same for you?

**Comment by gustavonalle** [comment]
@slaskawi Updated to download SHA1 from the repo

**Comment by infinispanrelease** [comment]
Started - TeamCity Infinispan :: Docker image Pull requests JDK8 [Build 26](http://ci.infinispan.org/viewLog.html?buildId=41139&buildTypeId=Infinispan_DockerImagePullRequestsJdk8&guest=1) for e218ac749f6b1fd57c51354b52b747aab18442da

**Comment by infinispanrelease** [comment]
**FAILURE** - TeamCity Infinispan :: Docker image Pull requests JDK8 [Build 26/gustavonalle:cluster](http://ci.infinispan.org/viewLog.html?buildId=41139&buildTypeId=Infinispan_DockerImagePullRequestsJdk8&guest=1) for e218ac749f6b1fd57c51354b52b747aab18442da
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:01:35

Process exited with code 1

Error message is logged

**Comment by infinispanrelease** [comment]
Started - TeamCity Infinispan :: Docker image Pull requests JDK8 [Build 27](http://ci.infinispan.org/viewLog.html?buildId=41141&buildTypeId=Infinispan_DockerImagePullRequestsJdk8&guest=1) for 70e4c544e80321ffeaede1a1a858327240a5c14f

**Comment by gustavonalle** [comment]
Updated with a fix to a typo in the documentation

**Comment by infinispanrelease** [comment]
**FAILURE** - TeamCity Infinispan :: Docker image Pull requests JDK8 [Build 27/gustavonalle:cluster](http://ci.infinispan.org/viewLog.html?buildId=41141&buildTypeId=Infinispan_DockerImagePullRequestsJdk8&guest=1) for 70e4c544e80321ffeaede1a1a858327240a5c14f
Tests: 0, 0 failed (0 new), 0 ignored. Build time: 00:01:34

Process exited with code 1

Error message is logged

File: server/Dockerfile (hunk 2 above)
Comment (slaskawi) [inline server/Dockerfile]: Yes, Maven Central is also faster for me. Probably our Nexus is located in the US...

**Comment by slaskawi** [comment]
LGTM - merging.

**Pull request merged**